epilepsy12/forms_folder/case_form.py

```diff
@@ -141,8 +200,37 @@ def clean_nhs_number(self):
                 raise ValidationError("NHS Number already taken!")
 
         # check NHS number is valid
-        validity = nhs_number.is_valid(formatted_nhs_number)
+        validity = nhs_number_package.is_valid(formatted_nhs_number)
         if validity:
             return formatted_nhs_number
         else:
             raise ValidationError(f"{formatted_nhs_number} is not a valid NHS number")
+
+    def clean(self):
+        cleaned_data = super().clean()
+        nhs_number = cleaned_data.get("nhs_number")
+        unique_reference_number = cleaned_data.get("unique_reference_number")
+        if self.is_jersey:
+            if unique_reference_number is None:
+                raise ValidationError("URN is a mandatory field.")
+            if self.instance.unique_reference_number:
+                # there is an existing URN
+                if unique_reference_number != self.instance.unique_reference_number:
+                    # the new URN does not match the one stored
+                    if Case.objects.filter(
```

Comment:
We should implement this as a unique constrant at the database level not as a cleaner

Reply:
absolutely.